daft/filesystem.py

```diff
@@ -214,58 +197,87 @@ def _resolve_path_and_filesystem(
-
-    # Use pyarrow utility to resolve filesystem and this particular path.
-    # If a non-None filesystem is provided to this utility, it will ensure that
-    # it is compatible with the provided path.
-    # A non-None filesystem will be provided if:
-    #  - a user-provided filesystem was passed to _resolve_paths_and_filesystem.
-    #  - a filesystem for the paths' protocol exists in the protocol -> fs cache.
-    #  - a filesystem was resolved for a previous path; i.e., filesystem is
-    #    guaranteed to be non-None for all but the first path.
-    try:
-        resolved_filesystem, resolved_path = _resolve_filesystem_and_path(path, filesystem)
-    except pa.lib.ArrowInvalid as e:
-        if "Unrecognized filesystem type in URI" in str(e):
-            # Fall back to fsspec.
-            protocol = get_protocol_from_path(path)
-            logger.debug(f"pyarrow doesn't support paths with protocol {protocol}, falling back to fsspec.")
-            try:
-                fsspec_fs_cls = get_filesystem_class(protocol)
-            except ValueError:
-                raise ValueError("pyarrow and fsspec don't recognize protocol {protocol} for path {path}.")
-            fsspec_fs = fsspec_fs_cls()
-            resolved_filesystem, resolved_path = _resolve_filesystem_and_path(path, fsspec_fs)
-        else:
-            raise
-
-    # If filesystem is fsspec HTTPFileSystem, the protocol/scheme of paths
-    # should not be unwrapped/removed, because HTTPFileSystem expects full file
-    # paths including protocol/scheme. This is different behavior compared to
-    # pyarrow filesystems.
-    if not _is_http_fs(resolved_filesystem):
-        resolved_path = _unwrap_protocol(resolved_path)
-
-    resolved_path = resolved_filesystem.normalize_path(resolved_path)
-    return resolved_path, resolved_filesystem
-
-
-def _is_http_fs(fs: FileSystem) -> bool:
-    """Returns whether the provided pyarrow filesystem is an HTTP filesystem."""
-    from fsspec.implementations.http import HTTPFileSystem
-
-    return (
-        isinstance(fs, PyFileSystem)
-        and isinstance(fs.handler, FSSpecHandler)
-        and isinstance(fs.handler.fs, HTTPFileSystem)
-    )
+    protocol = get_protocol_from_path(path)
+    translated_kwargs: dict[str, Any]
+
+    def _set_if_not_none(kwargs: dict[str, Any], key: str, val: Any | None):
+        """Helper method used when setting kwargs for pyarrow"""
+        if val is not None:
+            kwargs[key] = val
+
+    ###
+    # S3
+    ###
+    if protocol == "s3":
+        translated_kwargs = {}
+        if io_config is not None and io_config.s3 is not None:
+            s3_config = io_config.s3
+            _set_if_not_none(translated_kwargs, "endpoint_override", s3_config.endpoint_url)
+            _set_if_not_none(translated_kwargs, "access_key", s3_config.key_id)
+            _set_if_not_none(translated_kwargs, "secret_key", s3_config.access_key)
```

Comment:
This naming asymmetry is a bit spooky to me, any particular reason for it?

Reply:
I think we just have it named differently :( 

PyArrow:
1. `$AWS_ACCESS_KEY_ID`: `access_key`
2. `$AWS_SECRET_ACCESS_KEY`: `secret_key`

Daft:
1. `$AWS_ACCESS_KEY_ID`: `key_id`
2. `$AWS_SECRET_ACCESS_KEY`: `access_key`

Maybe @samster25 knows where we inherited our naming from?

Reply:
@jaychia Ah yeah, I based it off the aws sdk and boto3 suffix rather than what pyarrow did. What do you think we should use as the source or truth?

Reply:
Oh I wasn't proposing a change, just wondering where we got it from 😛